.github/workflows/ci.yml

```
@@ -3,166 +1,218 @@
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r config/requirements.txt
+          pip install -r config/requirements-dev.txt
+      
+      - name: Run tests with coverage
+        run: |
+          PYTHONPATH=src pytest tests/ -v --cov=src --cov-report=xml --cov-report=html
+      
+      - name: Upload coverage
+        if: matrix.python-version == '3.10' && github.event_name == 'pull_request'
+        uses: codecov/codecov-action@v3
```

actionlint (1.7.7)

the runner of "codecov/codecov-action@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)